mlx/coverity.py

```diff
@@ -417,33 +328,174 @@ def process_coverity_nodes(self, app, doctree, fromdocname):
+            env (sphinx.environment.BuildEnvironment): Sphinx' build environment.
+            docname (str): Name of the document in which the error occurred.
+        """
+        chart_labels = {}
+        combined_labels = {}
+        for label in labels:
+            attr_values = label.split('+')
+            for attr_val in attr_values:
+                if attr_val in chart_labels.keys():
+                    report_warning(env, "Attribute value '%s' should be unique in chart option." % attr_val, docname)
+                chart_labels[attr_val] = 0
+            if len(attr_values) > 1:
+                combined_labels[label] = attr_values
+        return chart_labels, combined_labels
+
+    def get_filtered_defects(self, node, env):
+        """ Fetch defects from suds using filters stored in the given CoverityDefect object.
+
+        Args:
+            node (CoverityDefect): CoverityDefect object with zero or more filters stored.
+            env (sphinx.environment.BuildEnvironment): Sphinx' build environment.
+
+        Returns:
+            (suds.sudsobject.mergedDefectsPageDataObj) Suds mergedDefectsPageDataObj object containing filtered defects.
+        """
+        report_info(env, 'obtaining defects... ', True)
+        defects = self.coverity_service.get_defects(self.project_name, self.stream, checker=node['checker'],
+                                                    impact=node['impact'], kind=node['kind'],
+                                                    classification=node['classification'], action=node['action'],
+                                                    component=node['component'], cwe=node['cwe'], cid=node['cid'])
+        report_info(env, "%d received" % (defects['totalNumberOfRecords']))
+        report_info(env, "building defects table and/or chart... ", True)
+        return defects
+
+    def get_filled_row(self, defect, columns, *args):
+        """ Goes through each column and decides if it is there or prints empty cell.
+
+        Args:
+            defect (suds.sudsobject.mergedDefectDataObj): Defect object from suds.
+            columns (list): List of column names (str).
+
+        Returns:
+            (nodes.row) Filled row node.
+        """
+        row = nodes.row()
+        for item_col in columns:
+            if item_col == 'CID':
+                # CID is default and even if it is in disregard
+                row += create_cell(str(defect['cid']),
+                                   url=self.coverity_service.get_defect_url(self.stream, str(defect['cid'])))
+            elif item_col == 'Location':
+                info = self.coverity_service.get_defect(str(defect['cid']),
+                                                        self.stream)
+                linenum = info[-1]['defectInstances'][-1]['events'][-1]['lineNumber']
+                row += create_cell("{}#L{}".format(defect['filePathname'], linenum))
+            elif item_col in self.column_map.keys():
```

Comment:
This checks CID again I think. Can you recheck? And "CID" in column_map is "Cid" so I am not sure.

Reply:
Because of the `if - elif`, the check for `CID` attribute is done before this line gets reached. Storing "Cid" in the dict was indeed wrong. I've now stored all column names in upper cases to make the checks case-insensitive. I don't see the need for case-sensitivity. This might save some headache for the absent-minded user. 52aa444b8ad614cde4f351504d7f8325d12793b8